backend-models/service-tests/showDuggaservice-test.txt

```diff
@@ -0,0 +1,393 @@
+update submitted dugga
+
+***
+Pre-req: 
+    !isset($grade) || !$grade > 1 &&
+    isset($dbpwd) && strcmp($hashpwd,$dbpwd) === 0
+***
+
+***
+Pre-req variables
+***
+$pre-values =
+    {
+        $aid = 999
+        $cid = 2
+        $hash = “dfg4zxc5”
+        $haspwd = “asdfasdf”
+        $moment = 2002
+        $answer = NULL
+    }
+
+    SQL-query: INSERT INTO userAnswer(aid,cid,hash,password,moment,useranswer) VALUES (pre-values[0], pre-values[1], pre-values[2], pre-values[3], pre-values[4], pre-values[5]);
+
+* Get newly created data
+    SELECT * WHERE hash=pre-values[2];
+
+***
+* Update information on userAnswer
+* send parameters input to showDuggaservice.php
+***
+send
+    {
+        $opt = SAVDU
+        $hash = “dfg4zxc5”
+        $haspwd = “asdfasdf”
+        $answer = “im updated”
+    }
+
+    SQL-query: "UPDATE userAnswer SET submitted=NOW(), useranswer=$answer, timesSubmitted=timesSubmitted+1 WHERE hash=$hash AND password=$hashpwd
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using hash
+***
+
+	SQL-query: DELETE FROM userAnswer WHERE hash = pre-values[2];
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #6 - 293 - 307
+submit dugga
+
+***
+Pre-req: !isset($grade) || !$grade > 1
+***
+
+***
+* send parameters input to showDuggaservice.php
+***
+    send
+    {
+        $opt = “SAVDU”
+        $cid = 2
+        $coursevers = 52432
+        $duggaid = 1
+        $moment = 2002
+        $variant = 3
+        $hash = “ghj1ghj2”
+        $haspwd = “asddasdd”
+        $answer = NULL
+    }
+
+    SQL-query INSERT INTO userAnswer (cid,quiz,moment,vers,variant,hash,password,timesSubmitted,timesAccessed,useranswer,submitted) VALUES(2, 1, 2002, 52432 , 3, “ghj1ghj2” , “asddasdd”,1 ,1 , NULL ,now());")
+
+***
+get newly created data
+***
+
+	SQL-query: SELECT hash FROM useranswer WHERE hash = $hash
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using hash
+***
+
+	SQL-query: DELETE FROM userAnswer WHERE hash = $hash;
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #7 (rows 319-324)
+super-view data from useranswer on hash
+
+***
+Pre-req: 
+    isSuperUser = TRUE
+    $hash != "UNK"
+***
+
+***
+* send parameters input to showDuggaservice.php
+***
+    send
+    {
+        $hash = hash
```

Comment:
what value does it have, do you mean that there is a string value as in hash, if that's the case which value?
if it's just suppose to be text it needs the ' "" ' .